resource-service-integration/src/test/java/uk/gov/gchq/palisade/integrationtests/resource/BasicPersistenceTest.java

```diff
@@ -0,0 +1,187 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.gchq.palisade.integrationtests.resource;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
+import org.springframework.cloud.openfeign.EnableFeignClients;
+import org.springframework.context.annotation.Import;
+import org.springframework.data.jpa.repository.config.EnableJpaRepositories;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.annotation.DirtiesContext.ClassMode;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.transaction.annotation.Transactional;
+
+import uk.gov.gchq.palisade.integrationtests.resource.config.ResourceTestConfiguration;
+import uk.gov.gchq.palisade.integrationtests.resource.web.ResourceClientWrapper;
+import uk.gov.gchq.palisade.resource.LeafResource;
+import uk.gov.gchq.palisade.resource.impl.DirectoryResource;
+import uk.gov.gchq.palisade.resource.impl.FileResource;
+import uk.gov.gchq.palisade.resource.impl.SystemResource;
+import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
+import uk.gov.gchq.palisade.service.resource.ResourceApplication;
+import uk.gov.gchq.palisade.service.resource.repository.JpaPersistenceLayer;
+import uk.gov.gchq.palisade.util.ResourceBuilder;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import static org.hamcrest.core.IsEqual.equalTo;
+import static org.junit.Assert.assertThat;
+
+@EnableFeignClients
+@RunWith(SpringRunner.class)
+@Import(ResourceTestConfiguration.class)
+@SpringBootTest(classes = ResourceApplication.class, webEnvironment = WebEnvironment.DEFINED_PORT)
+@EnableJpaRepositories(basePackages = {"uk.gov.gchq.palisade.service.resource.repository"})
+@DirtiesContext(classMode = ClassMode.AFTER_CLASS)
+public class BasicPersistenceTest {
+
+    @Autowired
+    private JpaPersistenceLayer persistenceLayer;
+
+    @Autowired
+    private ResourceClientWrapper client;
+
+    /**
+     * Scenario as follows, where (F)iles, (D)irectories and (S)ystems are annotated respectively
+     *
+     *     S
+     *     |
+     *     D
+     *   / | \
+     * F   F  F
+     */
+
+    private static final SimpleConnectionDetail DETAIL = new SimpleConnectionDetail().uri("test-data-service");
+    private static final String EMPLOYEE_FORMAT = "employee";
+    private static final String CLIENT_FORMAT = "client";
+    private static final String AVRO_TYPE = "avro";
+    private static final String CSV_TYPE = "csv";
```

Comment:
I think you've got the formats and types the wrong way round here.
The type should be the type of file (e.g. employee, client etc.)
The format should be the file format value (e.g. avro, csv etc.)
```suggestion
    private static final String CSV_FORMAT = "csv";
```
While the tests still work this brings them in line with how they would look within the resource-service.